src/compiler/crystal/program.cr

```diff
@@ -578,6 +578,10 @@ module Crystal
     end
 
     def add_def(node : Def)
+      if node.receiver.nil? && node.name.in?("super", "previous_def")
+        node.raise "top-level 'def #{node.name}' is not allowed due to no way to call this"
```

Comment:
```suggestion
        node.raise "top-level 'def #{node.name}' is not allowed"
```

Reply:
@Sija Why?

Reply:
IMO there's no need to go into such details - and if so they should be phrased in proper English.

Reply:
Thanks. You are right.